gradle-idea-configuration/src/main/java/com/palantir/gradle/ideaconfiguration/UpdateExternalDependenciesXml.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.ideaconfiguration;
+
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.palantir.gradle.ideaconfiguration.externaldependencies.ComponentXml;
+import com.palantir.gradle.ideaconfiguration.externaldependencies.PluginDependencyXml;
+import com.palantir.gradle.ideaconfiguration.externaldependencies.ProjectXml;
+import java.io.File;
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+public abstract class UpdateExternalDependenciesXml extends DefaultTask {
+    @Nested
+    public abstract SetProperty<PluginDependency> getDependencies();
+
+    @OutputFile
+    public abstract RegularFileProperty getOutputFile();
+
+    public UpdateExternalDependenciesXml() {
+        getOutputFile().set(getProject().file(".idea/externalDependencies.xml"));
+    }
+
+    @TaskAction
+    public final void updateXml() {
+        Set<PluginDependency> dependencies = getDependencies().get();
+        File outputFile = getOutputFile().get().getAsFile();
+
+        if (dependencies.isEmpty()) {
+            return;
+        }
+
+        XmlMapper xmlMapper = new XmlMapper();
```

Comment:
I typically define ObjectMapper instances as static final fields within a class. Additionally, it might be preferable to use the interface when creating instances, like so: ObjectMapper xmlMapper = new XmlMapper();.

Also, we should maybe consider including the input and output factories for the xmlMapper. We've encountered issues in the past when these were not specified, as seen [here](https://github.palantir.build/gradle-plugins/gradle-failure-reports/pull/22).